### PR TITLE
module_adpater: correct the way to get the scheduling period

### PR DESCRIPTION
**Pull request by CyKuei**
The scheduling period should be acquired from dev->pipeline->period

Signed-off-by: CY Kuei <cyk@waves.com>

### CODE REVIEW

**Comment by sofci** [comment]
Can one of the admins verify this patch?
> reply **test&nbsp;this&nbsp;please** to run this test once

**Comment by gkbldcig** [comment]
Can one of the admins verify this patch?

**Comment by lyakh** [comment]
test this please

File: src/audio/module_adapter/module/waves.c [hunk 1]
```diff
@@ -356,8 +356,8 @@ static int waves_effect_init(struct processing_module *mod)
 	waves_codec->o_format = waves_codec->i_format;
 
 	waves_codec->sample_size_in_bytes = sample_bytes;
-	waves_codec->buffer_bytes = mod->period_bytes;
-	waves_codec->buffer_samples = waves_codec->buffer_bytes / src_fmt->channels /
+	waves_codec->buffer_samples = src_fmt->rate * dev->pipeline->period / 1000000;
```
Comment (lyakh) [inline src/audio/module_adapter/module/waves.c:359]: maybe you could add a comment to explain that you're preparing a buffer for 1 period worth of data

Comment (paulstelian97) [reply src/audio/module_adapter/module/waves.c:359]: I really don't like this change. It's unclear and feels wrong, so a better explanation should be mandatory.

Comment (CyKuei) [reply src/audio/module_adapter/module/waves.c:361]: The comment was added. Hopefully, it is more clear.

File: src/audio/module_adapter/module/waves.c [hunk 2]
```diff
@@ -356,8 +356,10 @@ static int waves_effect_init(struct processing_module *mod)
 	waves_codec->o_format = waves_codec->i_format;
 
 	waves_codec->sample_size_in_bytes = sample_bytes;
-	waves_codec->buffer_bytes = mod->period_bytes;
-	waves_codec->buffer_samples = waves_codec->buffer_bytes / src_fmt->channels /
+	// Prepare a buffer for 1 period worth of data
+	// dev->pipeline->period is in us
+	waves_codec->buffer_samples = src_fmt->rate * dev->pipeline->period / 1000000;
```
Comment (dbaluta) [inline src/audio/module_adapter/module/waves.c:361]: Please use C style comments ```/* ... *```. Also, please fix typo in commit subject.

Comment (CyKuei) [reply src/audio/module_adapter/module/waves.c:361]: Updated.

File: src/audio/module_adapter/module/waves.c (hunk 1 above)
Comment (lgirdwood) [reply src/audio/module_adapter/module/waves.c:361]: @CyKuei thanks, it looks clear now to me at least.

**Review by lgirdwood — APPROVED** [review]
@barry-waves pls check and review.

**Comment by barry-waves** [comment]
approved this pull requestㄡ

File: src/audio/module_adapter/module/waves.c (hunk 1 above)
Comment (dbaluta) [reply src/audio/module_adapter/module/waves.c:361]: This is not the correct way to add multi line C comments. Please use something like this:

```
/* Prepare a buffer for 1 period worth of data
 * dev->pipeline->period stands for the scheduling period in us */

Comment (CyKuei) [reply src/audio/module_adapter/module/waves.c:362]: @dbaluta, modified. Thanks for the input.

**Review by dbaluta — APPROVED** [review]

**Comment by lgirdwood** [comment]
SOFCI TEST

**Pull request merged**